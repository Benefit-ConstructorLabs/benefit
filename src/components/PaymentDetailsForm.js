import React from 'react';
import '../../styles/components/payment-details-form.scss';

const PaymentDetailsForm = ({ createPaymentDetails, setCardInput, setExpDateInput, setCcvInput, cardNumber, expDate, ccv}) => {
  function handleCardChange(event){
  }
  function handleExpDateChange(event){
    setExpDateInput(event.target.value)
  }
  function handleCcvChange(event){
    setCcvInput(event.target.value)
  }
  
  return (
    <div className="payment-details-form">
      <h3>Set up payment</h3>
      <form className="payment-details-form__form" onSubmit={(event) => {
        event.preventDefault();
        createPaymentDetails()
      }}>
        <input
          type="text"
          placeholder="Card number"
          value={cardNumber}
          onChange={event => handleCardChange(event)}
        />
        <input
          type="date"
          placeholder="Expiry date"
          value={expDate}
          onChange={handleExpDateChange}
        />
        <input
          type="text"
          placeholder="CCV"
          value={ccv}
          onChange={handleCcvChange}
        />
        <p>Set up a donor account?</p>
        <p>Continue donating anonymously:</p>
        <button type="submit">Donate -Price- to -First Name-</button>
      </form>
    </div>
  )
};

export default PaymentDetailsForm;
