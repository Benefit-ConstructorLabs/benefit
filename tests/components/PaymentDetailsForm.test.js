import React from 'react';
import renderer from 'react-test-renderer';
import PaymentDetailsForm from '../../src/components/PaymentDetailsForm';

// It looks like you are trying to inject Stripe context outside of an Elements context. 
// Please be sure the component that calls createSource or createToken is withinan <Elements> component.

describe('DonationForm', () => {
  test.skip('should render correctly', () => {
    const tree = renderer
      .create(<PaymentDetailsForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
