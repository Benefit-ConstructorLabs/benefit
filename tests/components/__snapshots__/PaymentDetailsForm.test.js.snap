// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonationForm should render correctly 1`] = `
<div
  className="payment-details"
>
  <h2>
    Set up payment
  </h2>
  <p>
    Already have an account?
  </p>
  <button
    className="btn"
    type="button"
  >
    Login
  </button>
  <h3
    className="payment-details__heading"
  >
    Continue donating anonymously
  </h3>
  <form
    className="payment-details__form"
    onSubmit={[Function]}
  >
    <p>
      <input
        className="nolabel"
        onChange={[Function]}
        placeholder="Card number"
        type="text"
      />
    </p>
    <p>
      <input
        className="nolabel"
        onChange={[Function]}
        placeholder="Expiry date"
        type="date"
      />
    </p>
    <p>
      <input
        className="nolabel"
        onChange={[Function]}
        placeholder="CCV"
        type="text"
      />
    </p>
    <button
      className="btn btn__primary btn__submit"
      type="submit"
    >
      Donate Â£undefined to undefined
    </button>
  </form>
</div>
`;
